usaspending_api/etl/management/commands/load_table_from_delta.py

```diff
@@ -164,44 +175,199 @@ def handle(self, *args, **options):
+                self._write_to_postgres(
+                    spark, upsert_df, qualified_upsert_temp_table, postgres_schema_def, self.table_spec["column_names"]
+                )
+                self._write_to_postgres(
+                    spark, delete_df, qualified_delete_temp_table, delete_schema_def, list(delete_schema_def)
+                )
+
+                updated_incrementally = True
+
+        if not updated_incrementally:
+            temp_table_suffix = "temp"
+
+            qualified_temp_table = self._prepare_temp_table(postgres_temp_schema, temp_table_suffix)
+
+            # Read from Delta
+            df = spark.table(source_delta_table)
+
+            # Reset the sequence before load for a table if it exists
+            if self.options["reset_sequence"] and self.table_spec.get("postgres_seq_name"):
+                postgres_seq_last_value = self._set_sequence_value(self.table_spec["postgres_seq_name"])
+            else:
+                postgres_seq_last_value = None
+
+            self.logger.info(
+                f"LOAD (START): Loading full Delta table {source_delta_table} to {qualified_temp_table} using {strategy} "
+                f"strategy"
+            )
+            self._write_to_postgres(
+                spark,
+                df,
+                qualified_temp_table,
+                postgres_schema_def,
+                self.table_spec["column_names"],
+                postgres_seq_last_value,
+            )
+
+            self.logger.info(
+                f"LOAD (FINISH): Loaded full Delta table {source_delta_table} to {qualified_temp_table} using {strategy} "
+                f"strategy"
+            )
+
+            if self.qualified_postgres_table:
+                self.logger.info(
+                    f"Note: this has merely loaded the data from Delta. For various reasons, we've separated the"
+                    f" metadata portion of the table download to a separate script. If not already done so,"
+                    f" please run the following additional command to complete the process: "
+                    f" 'copy_table_metadata --source-table {self.qualified_postgres_table} --dest-table {qualified_temp_table}'."
+                )
+
+        if delta_table_load_version_key:
+            update_last_staging_load_version(delta_table_load_version_key, last_update_version)
+
+        # We're done with spark at this point
+        if spark_created_by_command:
+            spark.stop()
+
+    def _surpassed_update_threshold(self, spark: SparkSession, df: DataFrame) -> bool:
+        """
+        Determines whether the number of records in the incoming DF exceeds the threshold of allowable
+        changes when compared to total number records in the table being copied. The threshold percentage
+        is based on the `incremental_parameter` parameter for the job.
+
+        Args:
+            spark (SparkSession): the active SparkSession to work within
```

Comment:
I don't think there's anything wrong with having both a type hint and the argument's type in the docstring. This format of having the arguments and their type in the docstring is very common and the type hints help inside of an IDE so I feel like each of these have their place.

I do understand the duplication of information here though, but I don't think it adds an unnecessary overhead to maintenance. I'm always in favor of more documentation, but that's just my two cents.

Reply:
Most duplication of knowledge is bad. This just introduces additional things for future team members to maintain, it's unnecessary, and very simple to fix.

Reply:
> type hints help inside of an IDE

In the IDE if I hover over a method without duplicated types in the pydoc, I still can easily see the types expected. So I don't understand why this pydoc helps the IDE when the types already show up. Can you clarify @aguest-kc

Reply:
I didn't say that the docstring helps the IDE. I said the type hints help inside an IDE, but I think the Google-style docstrings can make sense for some people when they're reading through the description of a function or method.

Either way these are stylistic decisions and I think we should decide on a standard code style for things like this at some point and go forward with that decision. Otherwise we'll almost certainly have future requests for changes that revolve around style since everyone has their own preferences. If/when we decide things like that we can update our CONTRIBUTING.md file so we have the standard documented.

I have no issue with removing the datatype from the docstring. I was just pointing out that this docstring format is one of the most commonly used even when type hints are present.

Reply:
> I didn't say that the docstring helps the IDE. 

You're right, I forgot the "in" in my response. I think you know what I meant but I meant "helps in the IDE".

Anyway, my vote is to stop knowledge duplication.

Reply:
> I didn't say that the docstring helps the IDE. 

You're right, I forgot the "in" in my response. Apologies for the confusion. That said, I think it's clear what I meant, i.e. "helps in the IDE".

Anyway, my vote is to stop knowledge duplication with repeated type hints. I hope us developers follow the best practice of "knowledge duplication is bad" mentality even without having to document everything that would be considered knowledge duplication.

All this said, I believe I understand where you're coming from. I think another option, which hasn't been stated, is that we could remove the duck typing from the method signature, e.g. the arguments; and leave the type hints in the docstring. Thoughts?

Reply:
Between the two options, my vote would be for your original suggestion of removing the datatypes from the docstring, but leaving the type hints in the signature.

Reply:
Went ahead and removed these from the docstrings.